LuteBot/IO/KB/ActionManager.cs

```diff
@@ -139,14 +139,17 @@ private static void InputCommand(int noteId)
             /*
              * If the cursor is showing, it means the user is typing or doesn't have the required state to play lute.
              */
-            if (pci.flags == CURSOR_SHOWING)
-                return;
 
-            if (((Control.ModifierKeys & Keys.Control) == Keys.Control) ||
-                ((Control.ModifierKeys & Keys.Shift) == Keys.Shift) ||
-                (Control.ModifierKeys & Keys.Alt) == Keys.Alt)
-                return;
+            if (ConfigManager.GetBooleanProperty(PropertyItem.DontPlayNoteWhenRequired) == true)
+            {
+                if (pci.flags == CURSOR_SHOWING)
```

Comment:
I hope this feature gets merged soon!
Question, when user does not have required state (like typing), does it pause the bot from playing?
If not, it might be nice to pause the player so they can resume playing the song where they left off?

Reply:
At the start i only wanted to release a version that would reset the song after resuming it, i'm too lazy to make something good enough, i know it would be much better to do what you said.

Reply:
It i guess we could backtrack the call branch and put the if statement at a better place, a place where the music would be implicitly paused, but i'm lazy, will maybe do it later.

Reply:
I'll see what I can do!

Reply:
I've made progress but I need help...

I wonder if there's a way for the program to know if the console is open?
The problem is that the cursor also shows when the console is open (aka when the note is being entered). So even in your current version, if you add `Console.WriteLine(pci.flags == CURSOR_SHOWING);` right before the block, you'll see that there's a lot of false positives.

In your case, because it doesn't actually stop the player, it's less noticeable, but in my case, it actually pauses on every false positive. Is there a reliable way for me to detect whether the console is open? I took a look at `consoleOn` in the same file, does not look promising.

Reply:
I don't remember having any issues playing song tho

Reply:
Oh you meant with your changes

Reply:
I meant both. I have my own bard playlist that have some heavy weight songs. I'll see if there's a file in the dump folder in this repo that could demonstrate this.

If the song is not very complex (like no chords), then it won't be an issue because the ingame console would have closed long before the next note hits. I'm talking about cases where a note hits before the previous one closed the console.

Reply:
I used this method because it's not really intrusive, i could read the memory from the game to check if the chatbox is being used, but VAC would probably not be happy with it.

Reply:
[entertainer](https://github.com/MontyLeGueux/Lutebot-2/blob/master/dump/the_entertainer.mid) in the dump file should demonstrate the problem after the intro.

After adding `Console.WriteLine(pci.flags == CURSOR_SHOWING);` before the block, ideally it should be always false if users don't do anything. But when the song gets faster and more complex, you'll see a few trues in there.

Reply:
Ah, I forgot one thing to try. I'll try to set my own instance variable that turns on and off when the program sends the 'open console' command and 'enter' command, see how that goes.

Reply:
@MontyLeGueux 
I got a solution working. The main idea is knowing that console could still be up when `CURSORINFO()` first checks, we give it a very brief timeout (9ms is the smallest value that it works on my machine without false positives), and check it again. If the cursor is still up after the timeout, we assume it's user initiated.

here's an ugly snippet:
```C#
CURSORINFO pci = new CURSORINFO();
pci.cbSize = Marshal.SizeOf(typeof(CURSORINFO));
GetCursorInfo(ref pci); // it stores the cursordata to the struct

/*
    * If the cursor is showing, it means the user is typing or doesn't have the required state to play lute.
    */
if (ConfigManager.GetBooleanProperty(PropertyItem.DontPlayNoteWhenRequired) == true)
{
    if (pci.flags == CURSOR_SHOWING)
    {
        Thread.Sleep(9);
        CURSORINFO pci1 = new CURSORINFO();
        pci1.cbSize = Marshal.SizeOf(typeof(CURSORINFO));
        GetCursorInfo(ref pci1); // it stores the cursordata to the struct
        if (pci1.flags == CURSOR_SHOWING)
        {
            Console.WriteLine("Cursor still showing");
            return;
        }
    }
```

Reply:
I've opened a PR to your fork [here](https://github.com/ExtReMLapin/mordhau-Lutebot/pull/1)

Reply:
Thanks 👍  Merged all of it to this PR